### DISCUSSION TITLE
Configure Autograder failing in Gradescope - Issue with Mambaforge

### DISCUSSION THREAD

**Discussion by lebrown** [ucbds-infra/otter-grader#845] [Q&A] [answered]
I was updating an assignment's autograder on Gradescope and saw that the configure autograder step failed. 
Given, I had just made a minor tweak on the assignment, I was surprised. 

The error message was: 
"The command '/bin/sh -c bash /gradescope/setup.sh' returned a non-zero code: 127"

Specifically in the build output: 
```
Extracting mamba-1.5.9-py312h9460a1c_0.conda
!!!!!! Mambaforge is now deprecated !!!!!
Future Miniforge releases will NOT build Mambaforge installers.
We advise you switch to Miniforge at your earliest convenience.
More details at [https://conda-forge.org/news/2024/07/29/sunsetting-mambaforge/.](https://conda-forge.org/news/2024/07/29/sunsetting-mambaforge/)
If you are unable to switch to Miniforge, you may pin your installer version to one found in 
https://github.com/conda-forge/miniforge/releases/tag/24.3.0-1
or if you lack the system requirements (Linux glibc >= 2.17, or macOS + x86-64bit >= 10.13)
you may pin your installer to one older version found in 
https://github.com/conda-forge/miniforge/releases/tag/24.3.0-0
This Miniforge installer will ceese to work in 2025.
If you require this Mambaforge installer you may pin to 
https://github.com/conda-forge/miniforge/releases/tag/24.7.1-0
ERROR: executing pre_install.sh failed
/autograder/source/setup.sh: line 24: mamba: command not found
/autograder/source/setup.sh: line 27: mamba: command not found
```

It looks like the issue is with the sunsetting of mambaforge. 

Note, I am using an older version of otter, v5.1.3. 

Has this been updated in newer versions of otter? Or something that needs updating in `setup.sh`?
Any advice is appreciated!

**Comment by rlmarks** [comment]
In case you hadn't seen this potential workaround: https://github.com/ucbds-infra/otter-grader/issues/846
(note: i have not tried this yet, but have the same problem as above)

**Comment by chrispyles** [comment, marked as answer]
Moving discussion to #846. TL;DR: I have started #847 to fix this.

**Comment by lebrown** [comment]
`rlmarks` thanks for the quick response.   I was able to use your work around this afternoon. 

`chrispyles` thanks for the immediate fix!
